### PR TITLE
add common greenhouse gases as individual base units

### PR DESCRIPTION
**Pull request by danielhuppmann**
This PR adds the most common greenhouse gases as individual base units. It also adds a warning to the definitions-file not to add CO2-equivalents here - this should wait for a cleaner solution to use pint-contexts or a clear reference to the IPCC report and the GWP-x.

PS: I tried to have both carbon and carbon dioxide as base units like
```
carbon_dioxide = [carbon_dioxide] = co2 = CO2
carbon = [carbon] = C=
# molecular weight of carbon dioxide to that of carbon: 44/12
[carbon] = 44/12 * [carbon_dioxide]
```
but pint seems to ignore the factor in the dimensions-conversion...

### CODE REVIEW

**Comment by danielhuppmann** [comment]
maybe @znicholls and @Rlamboll you can take a quick look if the warning makes sense?

**Comment by znicholls** [comment]
The warning makes sense. My only concern is that we have a clean solution to this in scmdata, is the hesitancy to use it because it’s wrapped in a whole bunch of other stuff? If I move it to another repo will that make it easier?

**Comment by khaeru** [comment]
> we have a clean solution to this in scmdata

For anyone else reading this thread, I searched for the “scmdata” documentation and found this about units handling: https://scmdata.readthedocs.io/en/latest/scmdata.units.html

The units and factors seem to be embedded in this source file: https://github.com/lewisjared/scmdata/blob/master/src/scmdata/units.py

**Comment by khaeru** [comment]
My comments to @danielhuppmann, repeated here for the benefit of others:

- The Pint 'chemistry' context provides an example of the intended usage, at least from POV of pint's maintainer.
  - Documentation: https://pint.readthedocs.io/en/0.11/contexts.html#parameterized-contexts
  - Definitions: https://github.com/hgrecco/pint/blob/7206b5caac5b9228f44f54c7500bb24e831027cf/pint/default_en.txt#L758-L786
- Here, the [physical quantities] and (base units) are from SI: [mass] (gram); [substance] (mole); etc.
- To convert [mass] of, say, carbon, to [substance], one provides:
  - the context that defines the relationship between two physical quantities to be used, and
  - the molecular weight/mass, e.g. in g/mol.

In the case of GWP, thinking out loud:
- Warming *per se* is measured as [temperature] e.g. of global mean surface air.
  - It can also be measured as radiative forcing, e.g. [power] / [length]² (W/m²).
- These changes (measured in [temperature] or [power]/[length]²) are *caused by* (under certain assumptions) a certain amount of carbon.
- Carbon in turn can be measured in [mass] or [substance].
- So one way (not the common one, I'll get to that) to define a context would be like this. I am copying here the syntax used by Pint:
  ```
  [mass] -> [temperature]: value * gwp
  ```
- In this case, one supplies the `gwp` for carbon *per se*, i.e. in [mass]/[temperature], e.g. kelvin per gigatonne (the exact units don't matter).
- Other GHGs are converted to [temperature] by their own `gwp` parameter.
- In layman's thinking, GWP is roughly [mass] of "CO2 that would cause equivalent warming to a certain [mass] of CH4".
- i.e. these both have the *same* physical quantity [mass], but they describe different things:
  - One is a certain amount of CH4.
  - The other is a hypothetical amount of CO2 that would cause (…etc.)
- To convert [mass] of CH4 to [mass] of "CO2 that would cause (…etc.)", one could:
  1. Convert [mass] CH4 to [temperature] using the `gwp` for CH4.
  2. Convert [temperature] to [mass] using the `gwp` for CO2.
- This conversion is clear and explicit because the [temperature] is the *same* measurement of the *same* system.

I would prefer to use some variant of this approach, rather than what's in this commit, because it avoids abusing Pint by introducing physical quantities that are not actually physical quantities. (Using the definition “A physical quantity is a property of a material or system that can be quantified by measurement,” [mass] or [substance] (number of molecules) are properties of a system, e.g. a certain lump of carbon. [carbon] is the system itself.)

**Comment by danielhuppmann** [comment]
Thanks @khaeru & @znicholls for the comments - see a first round of responses...

0) the idea of the PR was to get a starting point, not tackle everything (warming, temperature)

1) sorry @znicholls, I indeed missed how many unit definitions you had already implemented in open-scm per [your comment](https://github.com/IAMconsortium/pyam/pull/341#pullrequestreview-369964970) - just looked at the implementation of the unit registry and found that more complicated than what was necessary for pyam

2) importing open-scm into pyam isn’t an option because of the circular dependency, plus we still wouldn’t have the benefit of using it across multiple applications (which is why we started this repo)

3) my intuition of context would have been something like “use the context (ie conversion factors) of AR4-GWP100 for converting CH4 to CO2”

**Comment by znicholls** [comment]
Thanks, just for quick clarification before I sign off for the night. The units are in scmdata, not OpenSCM. The offer of splitting out scmdata.units into its own repo (in the comments above) would solve the circular dependency. If you feel scmdata.units is too complicated then we won’t worry about shifting it around. We wanted to make it simpler but failed in that quest.

**Comment by khaeru** [comment]
Proof of concept:
https://github.com/IAMconsortium/units/blob/33e873639f086a8fe9d143d12c299b3ab86f78ef/test.py#L56-L79

**Comment by danielhuppmann** [comment]
force-pushed (with consent from @khaeru) to keep the context discussion for later

**Comment by danielhuppmann** [comment]
Ok, back to the roots on this PR after @khaeru opened a can of worms - let's move the discussion how best to integrate the work done in `openscm` to #9.

Here, let's implement just the changes so that users of this repo can do conversions like `Mt CO2 -> Gt co2`. This does not introduce any inconsistency with further integration with `openscm` later.

**Comment by danielhuppmann** [comment]
Added two checks of the new emissions species as requested by @khaeru (and fixed the tests with his help, thanks!), so good to go.

Writing the tests made me realise that `C` is already used by pint for `Coulomb`, so I removed that and will reintroduce it as part of the gwp-context effort, see #10 (so that we don't overwrite pint-default units)
